dali/operators/reader/loader/video_loader.cc

```diff
@@ -211,19 +212,76 @@ typename std::enable_if<!std::numeric_limits<T>::is_integer, bool>::type
         || std::abs(x-y) < std::numeric_limits<T>::min();
 }
 
+static int read_packet(void *opaque, uint8_t *buf, int buf_size) {
+  OpenFile::VideoFileDesc *file_data = static_cast<OpenFile::VideoFileDesc*>(opaque);
+
+  // open file if needed
+  if (!file_data->file_stream) {
+    file_data->file_stream = fopen(file_data->filename.c_str(), "r");
+    if (!file_data->file_stream) {
+      DALI_FAIL(std::string("Could not open file ") + file_data->filename);
+    }
+    fseek(file_data->file_stream, file_data->file_position, SEEK_SET);
+  }
+  return fread(buf, 1, buf_size, file_data->file_stream);
+}
+
+static int64_t seek_file(void *opaque, int64_t offset, int whence) {
+  OpenFile::VideoFileDesc *file_data = static_cast<OpenFile::VideoFileDesc*>(opaque);
+
+  // open file if needed
+  if (!file_data->file_stream) {
+    file_data->file_stream = fopen(file_data->filename.c_str(), "r");
+    if (!file_data->file_stream) {
+      DALI_FAIL(std::string("Could not open file ") + file_data->filename);
+    }
+    fseek(file_data->file_stream, file_data->file_position, SEEK_SET);
+  }
+  int ret = -1;
+  switch (whence) {
+    case SEEK_SET:
+    case SEEK_CUR:
+    case SEEK_END:
+      ret = fseek(file_data->file_stream, offset, whence);
+      break;
+    default:
+      break;
+  }
+  if (ret == 0) {
+    return ftell(file_data->file_stream);
+  } else {
+    return ret;
+  }
+}
+
 OpenFile& VideoLoader::get_or_open_file(const std::string &filename) {
   auto& file = open_files_[filename];
 
-  if (!file.open) {
+  if (!file.loaded) {
+    file.file_desc.filename = filename;
     LOG_LINE << "Opening file " << filename << std::endl;
 
-    AVFormatContext* raw_fmt_ctx = nullptr;
-    int ret = avformat_open_input(&raw_fmt_ctx, filename.c_str(), NULL, NULL);
+    AVFormatContext* raw_fmt_ctx = avformat_alloc_context();
```

Comment:
I'd create a temporary unique pointer with `make_unique_av` here to avoid leaking resources on failure.

Reply:
Done